### PR TITLE
Fix/graphql datasource normalization with retry

### PR DESCRIPTION
**Pull request by spetrunin**

### CODE REVIEW

File: pkg/ast/ast_object_type_definition.go [hunk 1]
```diff
@@ -88,3 +88,24 @@ func (d *Document) ImportObjectTypeDefinition(name, description string, fieldRef
 
 	return
 }
+
+func (d *Document) RemoveObjectTypeDefinition(name ByteSlice) bool {
```
Comment (pvormste) [inline pkg/ast/ast_object_type_definition.go:92]: I would recommend adding a dedicated test for this function

File: pkg/ast/ast_root_operation_type_definition.go [hunk 2]
```diff
@@ -74,12 +76,36 @@ func (d *Document) AddRootOperationTypeDefinition(rootOperationTypeDefinition Ro
...
 		NamedType: Type{
-			Name: d.Input.AppendInputString(name),
+			Name: d.Input.AppendInputBytes(nameBytes),
 		},
 	}
 
 	return d.AddRootOperationTypeDefinition(operationTypeDefinition)
 }
+
+func (d *Document) ReplaceRootOperationTypeDefinition(name string, operationType OperationType) (ref int) {
```
Comment (pvormste) [inline pkg/ast/ast_root_operation_type_definition.go:102]: test for this one too?

File: pkg/ast/index.go [hunk 3]
```diff
@@ -83,3 +87,20 @@ func (i *Index) FirstNodeByNameBytes(name []byte) (Node, bool) {
 	}
 	return node[0], true
 }
+
+func (i *Index) RemoveNodeByName(name []byte) {
```
Comment (pvormste) [inline pkg/ast/index.go:91]: and also a test maybe

**Review by pvormste — CHANGES_REQUESTED** [review]

**Comment by spetrunin** [comment]
tests will be done in a separate PR

**Pull request merged**